### PR TITLE
Fix issue with yahpo uploading/syncing with S3

### PR DESCRIPTION
**Pull request by geoalgo**
*Issue #404 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by wistuba — CHANGES_REQUESTED** [review]
It's not fully addressing the issue. Generation of the benchmark is still not happening.

**Comment by geoalgo** [comment]
> It's not fully addressing the issue. Generation of the benchmark is still not happening.

Could you be more specific? 

It is generating the benchmark in my case and is currently fixing an issue (local files were incorrectly checked for yahpo).

**Comment by wistuba** [comment]
When creating the issue, `generate()` was not triggered. I don't see how these changes will account for that. If you delete files from S3 and locally, does it recreate it correctly?

**Comment by wistuba** [comment]
For `load_blackbox("yahpo-rbv2_xgboost")` I get following exception:

```FileNotFoundError: [Errno 2] No such file or directory: '~/.blackbox-repository/yahpo/rbv2_xgboost/encoding.json'```

**Comment by geoalgo** [comment]
@wistuba it should work now, the logic with paths had to be updated to support sub directories on s3 (which is needed for yahpo).

File: syne_tune/blackbox_repository/repository.py [hunk 1]
```diff
@@ -98,20 +98,12 @@ def load_blackbox(
...
     """
-    if name.startswith("yahpo-"):
-        if yahpo_kwargs is None:
-            yahpo_kwargs = dict()
-        return instantiate_yahpo(name, **yahpo_kwargs)
-
-    tgt_folder = Path(repository_path) / name
+    tgt_folder = blackbox_local_path(name)
 
+    print(f"tgt folder {tgt_folder}")
```
Comment (wistuba) [inline syne_tune/blackbox_repository/repository.py:103]: remove

File: syne_tune/blackbox_repository/repository.py [hunk 2]
```diff
@@ -130,17 +122,19 @@ def load_blackbox(
             f"Skipping download of {name} as {tgt_folder} already exists, change skip_if_present to redownload"
         )
     else:
+        logging.info("Local files not found, attempting to copy from S3.")
         tgt_folder.mkdir(exist_ok=True, parents=True)
         try:
-            s3_folder = s3_blackbox_folder(s3_root)
+            s3_folder = blackbox_s3_path(name=name, s3_root=s3_root)
+            print(f"s3_folder {s3_folder}")
```
Comment (wistuba) [inline syne_tune/blackbox_repository/repository.py:129]: remove

File: syne_tune/blackbox_repository/repository.py (hunk 1 above)
Comment (geoalgo) [reply syne_tune/blackbox_repository/repository.py]: Ok, done.

**Review by wistuba — APPROVED** [review]

**Pull request merged**